docs/elasticsearch-cluster.yaml

```diff
@@ -0,0 +1,130 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: operator
+  namespace: es-operator-demo
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: es-config
+  namespace: es-operator-demo
+data:
+  elasticsearch.yml: |
+    cluster.name: es-cluster
+    network.host: "0.0.0.0"
+    bootstrap.memory_lock: false
+    discovery.zen.ping.unicast.hosts: es-cluster
+    discovery.zen.minimum_master_nodes: 1
+---
+apiVersion: apps/v1beta1
+kind: Deployment
+metadata:
+  name: es-master
+  namespace: es-operator-demo
+  annotations:
+    downscaler/exclude: "true"
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        application: elasticsearch
+        role: master
+    spec:
+      serviceAccountName: operator
+      securityContext:
+        fsGroup: 1000
+      initContainers:
+      - name: init-sysctl
+        image: busybox:1.30
+        command:
+        - sysctl
+        - -w
+        - vm.max_map_count=262144
+        resources:
+          requests:
+            memory: 50Mi
+            cpu: 50m
+          limits:
+            memory: 50Mi
+            cpu: 50m
+        securityContext:
+          privileged: true
+      containers:
+      - name: elasticsearch
+        resources:
+          requests:
+            memory: 600Mi
+            cpu: 100m
+          limits:
+            memory: 600Mi
+            cpu: 100m
+        securityContext:
+          runAsUser: 1000
+        image: docker.elastic.co/elasticsearch/elasticsearch-oss:6.6.0
+        env:
+        - name: "node.name"
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.name
+        - name: "ES_JAVA_OPTS"
+          value: "-Xmx300m -Xms300m"
+        - name: node.master
+          value: "true"
+        - name: node.data
+          value: "false"
+        readinessProbe:
+          httpGet:
+            scheme: HTTP
+            path: /_cluster/health?local=true
+            port: 9200
+          initialDelaySeconds: 5
```

Comment:
Hi, was looking at this to give it a try. Should this be something a little longer given you are starting up a JVM and ES?

Reply:
Yeah, probably we could give it 30 sec or so without a problem. Don't think we have really measured how fast it actually starts with these settings.

Reply:
I increased it to 20s. It doesn't make a big difference because it's for the readiness probe only.